### ISSUE TITLE
Error in Schema reading

### ISSUE THREAD

**Issue by Hsankesara** [RADAR-base/radarpipeline#36]
I was reading Radar questionnaire data and got this error. 

```bash
2022-09-23 14:48:35,086 [INFO]: Traceback (most recent call last):
  File "/Users/heetsankesara/work/RADAR/radar-pipeline/radarpipeline/radarpipeline/radarpipeline.py", line 22, in run
    project.fetch_data()
  File "/Users/heetsankesara/work/RADAR/radar-pipeline/radarpipeline/radarpipeline/project/project.py", line 385, in fetch_data
    self.data = SparkCSVDataReader(
  File "/Users/heetsankesara/work/RADAR/radar-pipeline/radarpipeline/radarpipeline/io/reader.py", line 104, in read_data
    schema = schema_reader.get_schema()
  File "/Users/heetsankesara/work/RADAR/radar-pipeline/radarpipeline/radarpipeline/io/reader.py", line 238, in get_schema
    field_type = self._get_field(typ)
  File "/Users/heetsankesara/work/RADAR/radar-pipeline/radarpipeline/radarpipeline/io/reader.py", line 264, in _get_field
    raise ValueError(f"Unknown data type: {data_type}")
ValueError: Unknown data type: ['null', 'double']
```

I think it is because the data type in that particular schema is `['null', 'double']` which means it can be either of them. The schema that caused this error:- https://github.com/RADAR-base/RADAR-Schemas/blob/master/commons/active/questionnaire/questionnaire.avsc

Related issue: #24

**Comment by thepushkarp** [comment]
Interesting. Seems like we'd have to make the Schema Reader module much more robust as it should be able to support much more complex schema types. And in cases of something unexpected, fail gracefully than getting into runtime exception.

**Comment by Hsankesara** [comment]
Agreed, especially we need to accommodate cases where a column has multiple data types. Seems like it is possible to do it in spark. https://stackoverflow.com/questions/71923139/specifying-column-with-multiple-datatypes-in-spark-schema

**Comment by thepushkarp** [comment]
The issue is that since multiple datatypes can be specified in the schema for a particular column (which means that the data in the column can have any one of the defined datatypes), we cannot determine the actual datatype for the column unless we have actually seen some of the data.

For example, if the schema says that some column **marks** can be either an `int` or a `double` then we cannot construct a schema for reading the data before actually seeing which among `int` or `double` the actual data is. The reason for this is that any column of a DataFrame can have only one datatype defined for it.

In our last meeting, we decided to go with the following approach to solve this issue:
- Keep a list of the multiple defined datatypes for a particular column from the schema file
- Infer the schema from the file (for which pySpark looks at a portion of the actual data)
- Match the datatype of the column from the inferred schema with the list of datatypes maintained from reading the schema file
- If the datatypes match, or if there is a way that they can be matched by typecasting the datatype of the inferred schema to the datatype of the schema file such that this **typecasting does not result in a loss of data**, then use the datatype
- Else, fallback to using a String datatype

**Comment by Hsankesara** [comment]
Solved in #45
